### PR TITLE
tests: add unitary tests and curl requests examples

### PR DESCRIPTION
**Pull request by bl4ckb0ne**
server: fix functions related to the tests

### CODE REVIEW

**Comment by binarytrails** [comment]
@Bl4ckb0ne First thing first, a PR is supposed to be one functionality that you what to add. Otherwise, it makes it hard to review all of the code for possible breaks etc. For now, we agreed for unit tests so remove the PEP8 commit and finish the incomplete unit tests of the REST server. It doesn't mean that we won't apply your tool to ensure PEP8 standards but for the latter explanations you should use this PR for unit tests only.

File: ring_api/wrappers/dring_cython.pyx [hunk 1]
```diff
@@ -445,9 +508,9 @@ cdef class Dring:
     cdef CallbacksClient cb_client
 
     def __cinit__(self):
-        self._FLAG_DEBUG          = dring_cpp.DRING_FLAG_DEBUG
-        self._FLAG_CONSOLE_LOG    = dring_cpp.DRING_FLAG_CONSOLE_LOG
-        self._FLAG_AUTOANSWER     = dring_cpp.DRING_FLAG_AUTOANSWER
+        self._FLAG_DEBUG = dring_cpp.DRING_FLAG_DEBUG
```
Comment (binarytrails) [inline ring_api/wrappers/dring_cython.pyx:545]: @Bl4ckb0ne I don't agree on this. It makes it hard to read the flags.

Comment (bl4ckb0ne) [inline ring_api/wrappers/dring_cython.pyx:545]: It's [pep8](https://www.python.org/dev/peps/pep-0008/#other-recommendations) standard

Comment (binarytrails) [inline ring_api/wrappers/dring_cython.pyx:545]: [Citation needed] -- they pretty much say to use common sense for readability.

File: ring_api/client.py [hunk 2]
```diff
@@ -28,56 +28,80 @@
...
         type='str', dest='host', default='127.0.0.1',
-        help='restful server host')
+        help='restful server host'
+    )
 
-    parser.add_option('--auto-answer',
+    parser.add_option(
+        '--auto-answer',
         action='store_true', dest='autoanswer', default=False,
-        help='force automatic answer to incoming call')
```
Comment (binarytrails) [inline ring_api/client.py:69]: @Bl4ckb0ne This just adds extra lines to make the file bigger, where did you read that?

File: ring_api/extra/servers/curl_requests.txt [hunk 3]
```diff
@@ -1,66 +0,0 @@
-GET: ?param=value
-POST: json
-
-/account/
-GET: curl http://localhost:8080/account/\?type\=SIP
-POST: curl -d '{"details":{"foo":"bar"}}' http://localhost:8080/account/
```
Comment (binarytrails) [inline ring_api/extra/servers/curl_requests.txt:6]: @Bl4ckb0ne I think it's a good idea to keep it for some folks that want to know how to craft curl requests. If it's incomplete, it is okay, just add a comment stating it. If you moved it, don't forget to recommit it!

File: ring_api/interfaces/configuration_manager.pxd [hunk 4]
```diff
@@ -32,28 +32,34 @@ from ring_api.interfaces.dring cimport *
...
+    void setAccountDetails(const string& accountID,
+                           const map[string, string]& details)
     void setAccountActive(const string& accountID, const boolean& active)
     map[string, string] getAccountTemplate(const string& accountType)
     string addAccount(const map[string, string]& details)
     void removeAccount(const string& accoundID)
     vector[string] getAccountList()
     uint64_t sendAccountTextMessage(const string& accountID, const string& to,
-        const map[string, string]& payloads)
+                                    const map[string, string]& payloads)
```
Comment (binarytrails) [inline ring_api/interfaces/configuration_manager.pxd:44]: @Bl4ckb0ne tabs or space madness?

File: ring_api/wrappers/dring_cython.pyx [hunk 5]
```diff
@@ -266,38 +337,54 @@ cdef class ConfigurationManager:
...
+        local       -- save the certificate in the local storage bool
 
         Return: list of ids of the pinned certificate
         """
-        return raw_list_to_list(confman_cpp.pinCertificate(certificate, local))
+            
+        #return raw_list_to_list(
+        cdef vector[uint8_t] raw_cert;
+        
+        #cert_list = list(certificate)
```
Comment (binarytrails) [inline ring_api/wrappers/dring_cython.pyx]: Don't forget to remove the dead code!

File: ring_api/server/flask/server.py [hunk 6]
```diff
@@ -84,84 +83,124 @@ def _init_api_resources(self):
...
-        self.api.add_resource(account.AccountsDetails,
+        self.api.add_resource(
+            account.AccountsDetails,
             '/accounts/<account_id>/details/',
-            resource_class_kwargs={'dring': self.dring})
+            resource_class_kwargs={'dring': self.dring}
+        )
+        
+        self.api.add_resource(
+            account.AccountsCipher,
```
Comment (binarytrails) [inline ring_api/server/flask/server.py]: AccountsCipher -> AccountsCiphers

**Comment by binarytrails** [comment]
@Bl4ckb0ne Great work! Tell me when you think it is finished. I shall merge. :octopus:

File: ring_api/wrappers/dring_cython.pyx [hunk 7]
```diff
@@ -266,38 +339,51 @@ cdef class ConfigurationManager:
...
 
         Return: list of ids of the pinned certificate
         """
-        return raw_list_to_list(confman_cpp.pinCertificate(certificate, local))
+
+        cdef vector[uint8_t] raw_cert
+
+        cert_list = [ord(x) for x in list(certificate)]
+
+        print(cert_list)
```
Comment (binarytrails) [inline ring_api/wrappers/dring_cython.pyx]: @Bl4ckb0ne Leftover?

**Pull request merged**